test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs

```diff
@@ -266,39 +311,60 @@ public void ModelVisitor_GeneratesCorrectLineMappings()
                                               IEnumerable<LineMapping> expectedLineMappings)
         {
             var inputFile = "TestFiles/Input/" + testName + ".cshtml";
-            var expectedCode = ReadResource("TestFiles/Output/DesignTime/" + testName + ".cs");
+            var outputFile = "TestFiles/Output/DesignTime/" + testName + ".cs";
+            var expectedCode = ResourceFile.ReadResource(_assembly, outputFile, sourceFile: false);
 
             // Act
             GeneratorResults results;
-            using (var stream = GetResourceStream(inputFile))
+            using (var stream = ResourceFile.GetResourceStream(_assembly, inputFile, sourceFile: true))
             {
                 results = host.GenerateCode(inputFile, stream);
             }
 
             // Assert
             Assert.True(results.Success);
-            Assert.Equal(expectedCode, results.GeneratedCode);
             Assert.Empty(results.ParserErrors);
-            Assert.Equal(expectedLineMappings, results.DesignTimeLineMappings);
-        }
 
-        private static string ReadResource(string resourceName)
-        {
-            using (var stream = GetResourceStream(resourceName))
+#if GENERATE_BASELINES
+            ResourceFile.UpdateFile(_assembly, outputFile, expectedCode, results.GeneratedCode);
+
+            Assert.NotNull(results.DesignTimeLineMappings); // Guard
```

Comment:
Should probably also compare the current line mappings so the user knows which tests are failing (if they run them all) to know which bits they need to copy n paste the line mappings for.